engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -97,27 +100,63 @@ class NextgenEngine @JvmOverloads constructor(
         }
 
         val executionContext = NadelExecutionContext(executionInput, serviceExecutionHooks)
+        val instrumentationContext = getInstrumentationContext(query, queryDocument, executionInput, instrumentationState)
 
-        // TODO: determine what to do with NQ
-        // instrumentation.beginExecute(NadelInstrumentationExecuteOperationParameters(query))
-
-        val results = coroutineScope {
-            fieldToService.getServicesForTopLevelFields(query)
-                .map { (field, service) ->
-                    async {
-                        try {
-                            executeTopLevelField(field, service, executionContext)
-                        } catch (e: Throwable) {
-                            when (e) {
-                                is GraphQLError -> newExecutionResult(error = e)
-                                else -> throw e
+        val result: ExecutionResult
```

Comment:
In Kotlin `if`, `when`, `try` etc are all expressions rather than statements i.e. they have a output value.

So to be more concise you should write

```kotlin
val result: ExecutionResult = try {
    mergeResults(
        …
    )
} catch (e) {
    instrumentation.onComplete(null, e)
    throw e
}
instrumentation.onComplete(result, null)
```

Reply:
very nice!